### ISSUE TITLE
DragonFly 4.6 and "Date parsing ran 81 tests ... 8 FAILED" 

### ISSUE THREAD

**Issue by noloader** [randombit/botan#943]
Testing on DragonFly 4.6, x86_64, GCC 4.9.4, Botan 2.0.1 (unreleased, revision git:e7e130c3d56eead0).

Configured with `./configure.py --cc=gcc --cc-bin=/usr/bin/g++`.

```
$ LD_LIBRARY_PATH=. ./botan-test
...
TLS Policy DH key verification ran 1 tests all ok
TLS Policy DSA key verification ran 2 tests all ok
TLS Policy ECDH key verification ran 2 tests all ok
TLS Policy ECDSA key verification ran 2 tests all ok
TLS Policy RSA key verification ran 2 tests all ok
TSS ran 2 tests all ok
Date parsing ran 81 tests in 0.19 msec 8 FAILED
Failure 1: Date parsing hour produced unexpected result '5' expected '0'
Failure 2: Date parsing hour produced unexpected result '18' expected '14'
Failure 3: Date parsing year produced unexpected result '2038' expected '2037'
Failure 4: Date parsing month produced unexpected result '1' expected '12'
Failure 5: Date parsing day produced unexpected result '1' expected '31'
Failure 6: Date parsing hour produced unexpected result '4' expected '23'
Failure 7: Date parsing hour produced unexpected result '5' expected '0'
Failure 8: Date parsing hour produced unexpected result '12' expected '8'
Note 1: Date parsing Test #1 failed
Note 2: Date parsing Test #2 failed
Note 3: Date parsing Test #3 failed
Note 4: Date parsing Test #1 failed
Note 5: Date parsing Test #2 failed
calendar_point::to_string ran 1 tests all ok
...
```

**Comment by randombit** [comment]
Thanks for the report. These look to be the same unexpected result as in #853  but additionally 

```
Failure 7: Date parsing hour produced unexpected result '5' expected '0'
Failure 8: Date parsing hour produced unexpected result '12' expected '8'
```

**Comment by randombit** [comment]
Should be fixed by #1336 since we no longer rely on the operating system for this computation
